src/lib/pickles/composition_types/composition_types.ml

```diff
@@ -68,24 +70,204 @@ module Wrap = struct
 
         module In_circuit = struct
           module Lookup = struct
-            type ('scalar_challenge, 'fp) t =
-              { joint_combiner : 'scalar_challenge
-              ; lookup_gate : 'fp
-                    (** scalar used on the lookup gate selector. *)
-              }
+            type 'scalar_challenge t = { joint_combiner : 'scalar_challenge }
             [@@deriving sexp, compare, yojson, hlist, hash, equal, fields]
 
-            let to_struct l = Hlist.HlistId.[ l.joint_combiner; l.lookup_gate ]
+            let to_struct l = Hlist.HlistId.[ l.joint_combiner ]
+
+            let of_struct Hlist.HlistId.[ joint_combiner ] = { joint_combiner }
 
-            let of_struct Hlist.HlistId.[ joint_combiner; lookup_gate ] =
-              { joint_combiner; lookup_gate }
+            let map ~f { joint_combiner } =
+              { joint_combiner = f joint_combiner }
 
-            let typ (type f fp) scalar_challenge
-                (scalar : (fp, _, f) Snarky_backendless.Typ.t) =
+            let typ (type f fp) scalar_challenge =
               Snarky_backendless.Typ.of_hlistable ~var_to_hlist:to_hlist
                 ~var_of_hlist:of_hlist ~value_to_hlist:to_hlist
-                ~value_of_hlist:of_hlist
-                [ scalar_challenge; scalar ]
+                ~value_of_hlist:of_hlist [ scalar_challenge ]
+          end
+
+          module Optional_column_scalars = struct
+            type 'fp t =
+              { chacha0 : 'fp
+              ; chacha1 : 'fp
+              ; chacha2 : 'fp
+              ; chacha_final : 'fp
+              ; range_check0 : 'fp
+              ; range_check1 : 'fp
+              ; foreign_field_add : 'fp
```

Comment:
If this one is here because it is meant to use a lookup table, then it should go because FFAdd uses none (as shown in the other PR)